app/controllers/users_controller.rb

```diff
@@ -15,6 +16,16 @@ def update
     end
   end
 
+  def delete_card
+    customer = Stripe::Customer.retrieve(params[:customer_id])
+    customer.sources.retrieve(params[:card_id]).delete
+
+    Card.find_by(stripe_card_id: params[:card_id]).destroy
```

Comment:
Find the card from `current_user`. Otherwise anyone can delete anyones card since we don't have authorisation. so `current_users.cards.find_by(stripe_card_id: params[:card_id]).destroy`